### PR TITLE
Fix Sizing

### LINKED ISSUE #266: Add more sizing options
We could probably at least add back S/L option?

### PR DESCRIPTION
**Pull request by TurtIeSocks**
Fixes #266 

- Nearly all filter menus now have S/M/L/XL
- All filters that have size options should resize properly now
- Consolidates icon resizing to one function for easier visibility
- Removes resizing from portals and verified timers 
- Pokemon filter buttons now maintain local IV and size options instead of resetting them (except for reset all)

image
Tiny rats can be adjusted separately from the fatties.

### CODE REVIEW

File: static/js/index.js [hunk 1]
```diff
@@ -2425,16 +2425,9 @@ function loadScanAreaPolygons () {
...
-        // TODO: console.log('Pokemon size undefined:', id, 'form:', pokemonForm);
-        return 40;
-    }
+    const id = pokemonForm === 0 ? `${pokemonId}-${masterfile.pokemon[pokemonId].default_form_id}` : `${pokemonId}-${pokemonForm}`;
     const size = pokemonFilter[id].size;
-    if (size === 'huge') {
-        return 75;
-    }
-    return 40;
+    return (size === 'small' ? 20 : size === 'large' ? 60 : size === 'huge' ? 80 : 40);
```
Comment (Mygod) [inline static/js/index.js:2430]: Minor complaint/suggestion: A switch/case or using an object as dictionary would be more readable.

File: static/js/index.js [hunk 2]
```diff
@@ -5646,6 +5627,31 @@ let setPokemonFilters = (type, show, filterInfo) => {
     $('#table-filter-pokemon').DataTable().rows().invalidate('data').draw(false);
 }
 
+let setPokemonSize = (size) => {
```
Comment (Mygod) [inline static/js/index.js:5630]: `const` prefered?

**Review by Mygod — COMMENTED** [review]
Some random comments, for now.

File: src/configs/default.json [hunk 3]
```diff
@@ -63,7 +63,19 @@
...
+            "raid": { "small": 20, "normal": 40, "large": 60, "huge": 80 },
+            "pokestop": { "small": 10, "normal": 30, "large": 50, "huge": 70 },
+            "quest": { "small": 10, "normal": 30, "large": 50, "huge": 70 },
+            "invasion": { "small": 10, "normal": 30, "large": 50, "huge": 70 },
+            "spawnpoint": { "small": 0.5, "normal": 1, "large": 2, "huge": 3 },
+            "pokemon": { "small": 20, "normal": 40, "large": 60, "huge": 80 },
+            "nest": { "small": 20, "normal": 40, "large": 60, "huge": 80 },
+            "device": { "small": 10, "normal": 30, "large": 50, "huge": 70 }
+        },
+        "portalMods": { "oldColor": "blue", "newColor": "green" }
```
Comment (Mygod) [inline src/configs/default.json:78]: Minor comment: could use some trailing commas.

File: static/js/index.js [hunk 4]
```diff
@@ -2508,47 +2487,22 @@ function getQuestIndex (questId) {
...
+    let filterId
+    let filterType = eval(`${type}Filter`)
+    switch (type) {
+        case 'gym':         filterId = `t${id}`; break;
+        case 'raid':        filterId = form === undefined ? `l${id}` : form === 0 ? `p${id}` : `p${id}-${form}`; break;
+        case 'pokestop':    filterId = id; break;
+        case 'quest':       filterId = id; break;
+        case 'invasion':    filterId = id; break;
+        case 'spawnpoint':  filterId = id; break;
+        case 'pokemon':     filterId = form === 0 ? `${id}-${masterfile.pokemon[id].default_form_id}` : `${id}-${form}`; break;
```
Comment (Mygod) [inline static/js/index.js:2500]: How about support for big-karp/small-rat?

**Review by Mygod — COMMENTED** [review]
Great PR! More minor comments.

File: static/js/index.js (hunk 4 above)
Comment (TurtIeSocks) [reply static/js/index.js:2500]: Added in 9032bd

**Review by Mygod — APPROVED** [review]

**Review by clburlison — APPROVED** [review]
Huge is comical for raids & pokemon. Everything works

**Review by versx — APPROVED** [review]

File: src/configs/default.json [hunk 5]
```diff
@@ -63,7 +63,19 @@
             "iv": { "value": 100, "color": "red" },
             "pvp": { "value": 1, "color": "blue" },
             "both": { "color": "purple" }
-        }
+        },
+        "iconSizes": {
```
Comment (versx) [inline src/configs/default.json:67]: Nice touch with the customizable icon sizes and custom portal colors. 😄

**Comment by clburlison** [comment]
👍  fixed the filtering issues I was seeing. Now this is ready to merge from my POV.

**Pull request merged**